### DISCUSSION TITLE
Isaac Lab Multi-Agent skrl documentation not working

### DISCUSSION THREAD

**Discussion by Saumya-Karan** [Toni-SM/skrl#238] [Q&A] [answered]
I tried running MAPPO to train the multi-agent environment `Isaac-Shadow-Hand-Over-Direct-v0` with skrl using the following command:
`isaaclab.bat -p source\standalone\workflows\skrl\train.py --task Isaac-Shadow-Hand-Over-Direct-v0 --headless --algorithm MAPPO`

I am getting the following error:

```
Error executing job with overrides: 
Traceback (most recent call last):
File “C:\Users\saumy\IsaacLabrepo\IsaacLab\source\extensions\omni.isaac.lab_tasks\omni\isaac\lab_tasks\utils\hydra.py”, line 99, in hydra_main
func(env_cfg, agent_cfg, *args, **kwargs)
File “C:\Users\saumy\IsaacLabrepo\IsaacLab\source\standalone\workflows\skrl\train.py”, line 178, in main
runner.run()
File “C:\Users\saumy\isaaclab\lib\site-packages\skrl\utils\runner\torch\runner.py”, line 376, in run
self._trainer.train()
File “C:\Users\saumy\isaaclab\lib\site-packages\skrl\trainers\torch\sequential.py”, line 84, in train
self.multi_agent_train()
File “C:\Users\saumy\isaaclab\lib\site-packages\skrl\trainers\torch\base.py”, line 293, in multi_agent_train
shared_states = self.env.state()
File “C:\Users\saumy\isaaclab\lib\site-packages\skrl\envs\wrappers\torch\isaaclab_envs.py”, line 136, in state
return self._env.state()
AttributeError: ‘OrderEnforcing’ object has no attribute ‘state’

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
2024-12-13 07:46:35 [14,426ms] [Warning] [carb] Recursive unloadAllPlugins() detected!
```

Please explain how to resolve.

**Comment by Toni-SM** [comment, marked as answer]
Hi @Saumya-Karan 

This is because `gymnasium-v1.0.0` deprecated direct access to the `DirectMARLEnv.state` method.

I am going to update next **skrl** release to take it into account.
Meanwhile, please use gymnasium-v0.29.1: `isaaclab.bat -p -m pip install "gymnasium==0.29.1"`

**Comment by Rishi-V** [comment]
I encountered a similar issue using --algorithm=IPPO. 
When I looked at the `Trainer.multi_agent_train(self)` function in `skrl/trainers/torch/base.py` I saw that the only places that env.state() are used are for the `infos` dictionary. So my hacky current solution was to comment out the few lines referring to env.state() and shared_states and I am able to get IPPO running.

I plan to create several custom MARL environments with Isaac Lab this new year (Happy New Years!) so looking forward to an official fix. 🙏

PS: Thanks for writing this codebase in the first place! 🙂

**Comment by Toni-SM** [comment]
Fixed in version 1.4.0 (public now)
